Source/WebCore/dom/Document.cpp

```diff
@@ -3578,8 +3578,11 @@ const URL& Document::urlForBindings() const
         if (m_url.url().isEmpty() || !loader() || !isTopDocument() || !frame())
             return false;
 
-        auto* topDocumentLoader = topDocument().loader();
-        if (!topDocumentLoader || !topDocumentLoader->networkConnectionIntegrityPolicy().contains(NetworkConnectionIntegrity::Enabled))
+        auto* targetLoader = topDocument().loader();
+        if (targetLoader && LegacySchemeRegistry::isUserExtensionScheme(targetLoader->request().url().protocol()) && !LegacySchemeRegistry::isUserExtensionScheme(url().protocol()))
```

Comment:
We should not use LegacySchemeRegistry::isUserExtensionScheme, and we should be working to remove it.  Use URL::protocolIsInHTTPFamily instead.

Reply:
Changed this to use URL::hasSpecialScheme().